### DISCUSSION TITLE
Web Api On behalf Authentication

### DISCUSSION THREAD

**Discussion by ChrTall** [pnp/pnpcore#649] [Q&A]
I have written a Web Api with the help of PnP Core Sdk that exposes some Endpoints which at the moment are not secured. (I registered the app with App permissions.)
The use case is that it should be only called from our Business Central Application. Can you give me some advice if my plan for Authentication is the right one?
I want to register the app with on behalf permissions so that the bearer token is sent in the header when a BC user calls the api. I saw the Azure functions blog post, that is linked in the documentation.

**Comment by jansenbe** [comment]
@ChrTall : using the on-behalf-of flow is indeed the best option. If your user in authenticated to BC (assuming that's also Azure AD secured) then using the on-behalf-of flow the user can call into the Azure function and the Azure function can act as the user when it talks to SharePoint Online. The blog post from @s-KaiNet nicely describes this pattern: https://spblog.net/post/2020/12/10/how-to-access-sharepoint-data-from-azure-function-with-spfx-and-pnp-core
